update proposerConfig related sections (#409)
ConsenSys/doc.teku: docs/Reference/CLI/CLI-Syntax.md

Proposer configuration attributes:

* `fee_recipient` - (optional in `proposal_config` but required in `default_config`) [Fee recipient](../../HowTo/Prepare-for-The-Merge.md#configure-the-fee-recipient)
  to be used when proposing blocks.
* `builder` - (optional) Includes three attributes:
    * `enabled` - (optional in `proposal_config` but required in `default_config`) Indicates whether
      to use the [builder endpoint](#builder-endpoint) when proposing blocks.
      The default is `false`.
    * `gas_limit` - (optional) Gas limit for the builder.
      The default is `30000000`.
    * `registration_overrides` - (optional) Dedicated overrides to be used during the registration process.
      Useful for `DVT` and `SSV` technologies.
      The override is specified using the following attributes:
        * `timestamp` - (optional) Timestamp to be used (instead of the current time) in the validator
          registration message.
        * `public_key` - (optional in `proposal_config` but forbidden in `default_config`) Public
          key to be used (instead of the validator's public key) in the validator registration message.

> Each attribute value, for a given validator key, will be determined following the priority:

Reviewer: ```suggestion
Each attribute value, for a given validator key, is determined using the following priority:
```

> 1. specific configuration in `proposer_config`
> 1. default configuration `default_config`
> 1. default CLI argument (applicable only to `builder.enabled`)
> 1. default value (applicable only to `builder.enabled`)

Reviewer: ```suggestion
1. Specific configuration in `proposer_config`
1. Default configuration in `default_config`
1. Default CLI argument (applicable only to `builder.enabled`)
1. Default value (applicable only to `builder.enabled`)
```

          "fee_recipient": "0x50155530FCE8a85ec7055A5F8b2bE214B3DaeFd3",
          "builder": {
            "gas_limit": "35000000"
          }
        },
        "0xa99a76ed7796f7be22d5b7e85deeb7c5677e88e511e0b337618f8c4eb61349b4bf2d153f649f7b53359fe8b94a38e44c": {
          "builder": {
            "enabled": false
          }
        },
      },
      "default_config": {
        "fee_recipient": "0x6e35733c5af9B61374A128e6F85f553aF09ff89A",
      }
    }
    ```

    * If `--validators-builder-registration-default-enabled` is specified in CLI

    Validator `0xa0578...` is configured as:
    
    ```json
    "fee_recipient": "0x50155530FCE8a85ec7055A5F8b2bE214B3DaeFd3",
    "builder": {
      "enabled": true,
      "gas_limit": "35000000"
    }
    ```

    Validator `0xa99a7...` is configured as:

    ```json
    "fee_recipient": "0x6e35733c5af9B61374A128e6F85f553aF09ff89A"
    "builder": {
      "enabled": false,
      "gas_limit": "30000000"
    }
    ```

    All other validators are configured as:
    
    ```json
    "fee_recipient": "0x6e35733c5af9B61374A128e6F85f553aF09ff89A",
    "builder": {
      "enabled": true,
      "gas_limit": "30000000"
    }
    ```

>     * If `--validators-builder-registration-default-enabled` is NOT specified (or set to false) in CLI

Author: @alexandratran the UI isn't that good for those IMO. Don't know if you have other ways to represent these two "* if .." contitions, otherwise we can merge.
